### ISSUE TITLE
Add DNS lookup and connect with port number

### ISSUE THREAD

**Issue by 45drive45** [anselal/antminer-monitor#23]
Enable DNS lookup so that this awesome software can connect via the internet to miners that have port forwarding enabled on firewalls

Example:
Instead of IP, we can state miner01.mycompany.net:8081
I assume that this monitor is connecting via port 80, so all we have to do is portforward port 80 on our router.
Then this would be a super super awesome software as you can monitor remotely your mining farm.

**Comment by gnanet** [comment]
@45drive45 monitoring solutions like this one are accessing the API of cgminer, which defaults to port 4028. Forwarding the web-gui port is not a good idea, because if someone can bruteforce, or wiretap your user/pass, then he gains full access, while the API allows a fine control of access rules to the specific areas, and differentiates between read and write operations.

Anyways if you have a set-up with remote miners, port-forwarding the API should be done in connection with IP-restrictions on the firewall, and you should place the monitoring solution on a fixed-IP host, that is the only IP allowed to pass trough the firewall.

Also reading the code, i assume you can refer to the miner with hostname or IP address (it may be the case that input field is validated against IP address, but the CgminerAPI class itself does not limit you to IP-only)

**Comment by anselal** [comment]
@gnanet Thnx for answering to our friend !! Actually you can restrict the access of the Cgminer API to only one address if you want ;)

@45drive45 From what you say I conclude that you didn't run my app. The app listens to port 5000 by default which you can change of course. As @gnanet already said it is not a good idea to forward any connections directly to your miners. I think you understand that this is a major security leak. What you can do though is to forward port 5000 to my app, this way you can monitor you miners safely. At this time the app only lets you see some stats and not control your miners so it is a much more safer solution than to expose your miner to the internet.

About the DNS lookup there is a way of adding the miners in your app using hostnames. Just edit your `hosts` file in the machine running my software and add entries like `192.168.0.3 AntminerS9_1` and then enter `AntminerS9_1` in the IP field of the. I already tested it and it words just fine.

### Summing Up
Run the software in a machine in the same network as your miners. Prefer a RaspberryPi if you are familiar with Linux. The Raspberryi is a low energy Single Board Computer (SBC). Setup a DynamicDNS if you don't have a static public IP so you can access your router. Forward port 5000 to the RPi to monitor your farm using AntminerMonitor. If you need help with the setup send me an email and I will help you with that.

I think we covered this issue and we can close it unless you want to ask anything else related to that matter.

**Comment by 45drive45** [comment]
Hi,

I did run your app locally and also tried remotely. I really like the information presented. Thank you very much for both your advice. 
I will try out the Rasberry pi advice.
